presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -550,6 +552,16 @@ public void setSetSchema(String schema)
         setSchema.set(requireNonNull(schema, "schema is null"));
     }
 
+    public void setSetPath(String path)
+    {
+        setPath.set(requireNonNull(path, "path is null"));
```

Comment:
nit: I know that in other parts of this class we put the requireNonNull calls inside another method call, but let's not do it here for the sake of slightly better readability:

```
requireNonNull(path, "path is null");
setPath.set(path);
```